### PR TITLE
Board pwm support

### PR DESCRIPTION
**Pull request by grgustaf**

### CODE REVIEW

**Comment by grgustaf** [comment]
The JrS modules patch broke the WebBluetoothDemo.js sample, so @brianjjones is looking into that.

When I backported this to before that patch, it worked fine, so I'm confident in my changes there. So this could perhaps go forward without that fix, or if it comes soon I can rebase on top of that and verify again.

File: samples/PWM.js [hunk 1]
```diff
@@ -1,34 +1,44 @@
...
 
-// set timings in milliseconds
-led0.setMilliseconds(1500, 1000);
+// set brightness to 66% using hw cycle-based values
+var led0 = pwm.open(0);  // same as "0", "PWM0", "IO3", or "PWM_0.0"
+led0.setCycles(100, 66);
 
 // set brightness to 33% using hw cycle-based values
-var led1 = pwm.open({channel: pins.IO5, period: 3, pulseWidth: 1});
+var led1 = pwm.open("1");  // same as 1, "PWM1", "IO5", or "PWM_0.1"
```
Comment (jimmy-huang) [inline samples/PWM.js]: it would be nice to use single ' instead of " for consistency.

File: src/zjs_board.h [hunk 2]
```diff
@@ -8,30 +8,48 @@
...
-} zjs_pin_t;
 
-// return codes from zjs_board_find_pin
-#define FIND_PIN_FAILURE -1
-#define FIND_PIN_INVALID -2
+#ifdef BUILD_MODULE_AIO
+/**
+ * Finds the Zephyr pin corresponding to a JavaScript AIO pin name or number
+ *
+ * @param jspin        JrSds number or string pin name
```
Comment (jimmy-huang) [inline src/zjs_board.h]: JrSds? typo?

File: src/zjs_board.c [hunk 3]
```diff
@@ -134,19 +185,43 @@ typedef struct {
...
+    //             validate the pin, otherwise one based on the prefix; when the
+    //             pin is found, remaps it to another pin if remap is specified;
+    //             finally returns the right device name in device-name, and
+    //             returns the correct Zephyr pin
+    ZJS_ASSERT(name, "name is NULL");
+    ZJS_ASSERT(default_range, "default_range is NULL");
+    ZJS_ASSERT(device_prefix, "device_prefix is NULL");
+
+    if (strnlen(name, NAMED_PIN_MAX_LEN) == NAMED_PIN_MAX_LEN) {
+        DBG_PRINT("pin name too long");
```
Comment (jimmy-huang) [inline src/zjs_board.c]: add \n

File: src/zjs_board.c [hunk 4]
```diff
@@ -158,115 +233,159 @@ const zjs_pin_t *zjs_find_pin(const char *name)
...
-                return FIND_PIN_FAILURE;
-            }
-            *pin_num = pin_desc->gpio;
-            return 0;
-        }
+    jerry_size_t size = FULL_PIN_MAX_LEN;
+    char name[size];
+    zjs_copy_jstring(jspin, name, &size);
+    if (!size) {
+        DBG_PRINT("pin name too long");
```
Comment (jimmy-huang) [inline src/zjs_board.c]: add \n

**Comment by jimmy-huang** [comment]
+1

**Pull request merged**